internal/broker/broker.go

```diff
@@ -618,7 +618,7 @@ func (b *Broker) handleIsAuthenticated(ctx context.Context, session *session, au
 			authInfo, err = b.refreshToken(ctx, session.oauth2Config, authInfo)
 			var retrieveErr *oauth2.RetrieveError
 			if errors.As(err, &retrieveErr) && b.provider.IsTokenExpiredError(*retrieveErr) {
-				// The refresh token is expired, so the user needs to authenticate via OIDC again.
+				log.Infof(context.Background(), "Refresh token expired for user %q, requesting new device authentication", session.username)
```

Comment:
Since we now support messages also in next case, can't we show the user a message too?

Reply:
Oh, sorry I missed the comment... But yes, it works in that way indeed... We need to double-check if GDM we've in stable supports it, though before releasing.

So add to your TO-CHECK before broker release todo list, please :)

Reply:
I already tested it on 24.04. I'm not sure if it makes sense to add this to the manual tests we do before release:
1. To trigger it (without patching the broker), we need an expired refresh token, which I don't think we can acquire unless we have a 3 month old token (I don't).
2. I don't expect that this GDM feature breaks for a specific Ubuntu release if it worked for that release before (in other words: now that I tested 24.04, I don't expect it to break in 24.04, but maybe in 25.10 or 26.04), so IMO we should test it before Ubuntu releases, but not before each broker release.

Ideally, we should have an e2e test which tests this, but again this requires an expired token which I don't have :/